File: app/src/main/java/com/stedi/randomimagegenerator/app/di/components/HomeComponent.java
package com.stedi.randomimagegenerator.app.di.components;

import com.stedi.randomimagegenerator.app.di.modules.HomeModule;
import com.stedi.randomimagegenerator.app.view.activity.HomeActivity;

import dagger.Subcomponent;

@Subcomponent(modules = HomeModule.class)
public interface HomeComponent {
    void inject(HomeActivity homeActivity);
}

File: app/src/main/java/com/stedi/randomimagegenerator/app/di/components/HomeComponent.kt
package com.stedi.randomimagegenerator.app.di.components

import com.stedi.randomimagegenerator.app.di.modules.HomeModule
import com.stedi.randomimagegenerator.app.view.activity.HomeActivityModel
import dagger.Subcomponent

@Subcomponent(modules = [(HomeModule::class)])
interface HomeComponent {
    fun inject(homeModel: HomeActivityModel)
}